akka-stream/src/main/scala/akka/stream/SubscriptionWithCancelException.scala

```diff
@@ -19,6 +20,12 @@ trait SubscriptionWithCancelException extends Subscription {
   def cancel(cause: Throwable): Unit
 }
 object SubscriptionWithCancelException {
-  case object NoMoreElementsNeeded extends RuntimeException with NoStackTrace
-  case object StageWasCompleted extends RuntimeException with NoStackTrace
+
+  /**
+   * Not for user extension
+   */
+  @DoNotInherit
+  sealed trait NonFailureCancellation
```

Comment:
```suggestion
  sealed trait NonFailureCancellation extends RuntimeException with NoStackTrace
```

?

Reply:
I think it's a good idea to have a common subclass. I'm not even sure if `StageWasCompleted` is really needed.

Reply:
Can't be a trait then, I'll change it.